AD support

tvdijen opened the pull request.

dlundgren commented:
I'd like to suggest against using force-pushes. I had history issues yesterday when I went to merge the recent changes in to my fork. I had to delete it and then re-fork.

tvdijen commented:
Yeah, my bad.. I have a tendency to do that.. I've fixed a minor thing and it wasn't worth a new tag, so I force-pushed and moved the tag as well :see_no_evil: I always tell myself if I do it quick enough, no one will notice :speak_no_evil:

My only concern right now is a couple of Psalm-issues that seem hard to fix..

tvdijen commented:
Note:  Failing test was fixed upstream in the SSP master-branch.. It can be safely ignored
